File: analyze_images.js

const fs = require('fs');
const path = require('path');

const featureDataPath = path.join(process.cwd(), 'src/data/featureData.tsx');
const content = fs.readFileSync(featureDataPath, 'utf8');

const imageRegex = /image:\s*["']([^"']+)["']/g;
let match;
const usages = {};
let totalSlots = 0;

while ((match = imageRegex.exec(content)) !== null) {
    const img = match[1];
    usages[img] = (usages[img] || 0) + 1;
    totalSlots++;
}

console.log(`Total image slots: ${totalSlots}`);
console.log('Duplicate usage:');
Object.entries(usages).forEach(([img, count]) => {
    if (count > 1) {
        console.log(`${img}: ${count}`);
    }
});

const availableImages = fs.readdirSync(path.join(process.cwd(), 'public/images'));
console.log(`\nAvailable images in public/images: ${availableImages.length}`);

File: analyze_images.py

import re
import os
from collections import Counter

file_path = 'src/data/featureData.tsx'
try:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # regex for image: "..." or image: '...'
    matches = re.findall(r'image:\s*["\']([^"\']+)["\']', content)
    
    total_slots = len(matches)
    counts = Counter(matches)
    duplicates = {k: v for k, v in counts.items() if v > 1}
    
    print(f"Total slots: {total_slots}")
    print(f"Unique images used: {len(counts)}")
    print(f"Duplicates found: {len(duplicates)}")
    print("Duplicate details:")
    for img, count in duplicates.items():
        print(f"{img}: {count}")

except Exception as e:
    print(f"Error: {e}")
